golos.social/golos.social.cpp

```diff
@@ -0,0 +1,139 @@
+#include "golos.social.hpp"
+#include <eosiolib/crypto.h>
+#include <common/hash64.hpp>
+
+namespace golos {
+
+using namespace eosio;
+
+extern "C" {
+    void apply(uint64_t receiver, uint64_t code, uint64_t action) {
+        social(receiver).apply(code, action);
+    }
+}
+
+social::social(account_name self)
+    : contract(self)
+{}
+
+void social::apply(uint64_t code, uint64_t action) {
+    if (N(follow) == action) {
+        execute_action(this, &social::follow);
+    }
+    if (N(unfollow) == action) {
+        execute_action(this, &social::unfollow);
+    }
+    if (N(block) == action) {
+        execute_action(this, &social::block);
+    }
+    if (N(unblock) == action) {
+        execute_action(this, &social::unblock);
+    }
+}
+
+void social::follow_or_unfollow(account_name follower, account_name following, bool follow) {
```

Comment:
@s-medvedev See golos.publication: upvote/downvote/unvote -> set_vote.
I will open issue to simplify this bad code, ok?

Reply:
No, it's different cases. upvote/downvote/unvote it's a good example when doing so is justified. In this case weight doesn't affect on logic. In case of block/unblock bool argument `block` affects to logic: one peace of code executes when `block` is true, and another - when false.